[IMP] run_pylint: Add feature to check new conventions just in modules changed.

moylop260 opened the pull request:
Currently, we have a problem that if we add a new conventions in pylint-test-1 we will have all repos red in travis builds for stable branches and PR ones.

But currently we are working with next steps:
 1. We have created a new pr to oca/PROJECT-X branch 8.0
 2. We received feedabck from reviewers:
   3. Please, rename the class to `CamelCase`
   4. Please, add OCA author.
   5. Please, change your comment `encondig` by `coding`
   6. Other ones.

If we can automate it, Why not automate it?
Save time of reviewer, save time of developers just in new changes.

Then we need a middle step to add new conventions without have all repos red.
We can change the modules progressively of new conventions.

This feature run with original
`pylint pylint.cfg all_modules`
plus run other one with new conventions in just modified modules
`pylint pylint_PR.cfg modified_modules`

New conventions by example

 - CamelCase

We don't need wait to fix all repositories in all base branches to add this new convention.
We can add new conventions and check it just in new changes.
After a consider time we can move new conventions to old conventions.

![travis_feature](https://cloud.githubusercontent.com/assets/6644187/9338522/c3073c18-45aa-11e5-8533-29cc0f61299a.png)

Notice in `TRAVIS_PULL_REQUEST='true'` case the erros are old_conventions + new_conventions.

The old conventions are active in all modules in PR cases.

Why `TRAVIS_PULL_REQUEST="true"` and `TRAVIS_BRANCH`?
We don't have control on travis app :(
TRAVIS_PULL_REQUEST is a [environment variable of travis](http://docs.travis-ci.com/user/environment-variables/#Convenience-Variables)
![environment variables - travis ci 2015-08-21 09-41-33](https://cloud.githubusercontent.com/assets/6644187/9410958/d3e63378-47e8-11e5-85ce-b64515c24c43.png)
This is needed here because we need to know if [is_pull_request](https://github.com/OCA/maintainer-quality-tools/blob/87f2b269667c02af210493e929eb9524eb455907/travis/travis_run_tests#L57-L58) (Currently we have this code in stable branch) to verify new conventions just in PR cases to detect a diff with base branch `TRAVIS_BRANCH`.
![environment variables - travis ci 2015-08-21 09-43-57](https://cloud.githubusercontent.com/assets/6644187/9411021/29e8fea4-47e9-11e5-82e1-25269c279a18.png)

Now we can get git diff of `TRAVIS_BUILD_DIR`
![environment variables - travis ci 2015-08-21 09-45-24](https://cloud.githubusercontent.com/assets/6644187/9411044/5d114688-47e9-11e5-8036-a53b2d53e20d.png)

With next command if is PR:
```python
  git --git_dir=${TRAVIS_BUILD_DIR}/.git diff-index origin/${TRAVIS_BRANCH}
```
Now, we have all items changed in a PR...

pedrobaeza commented:
As stated before, I don't think this is a good idea, because there can be non-modified modules that depends on the modified ones that get broken in the PR.

moylop260 commented:
@pedrobaeza 
This feature run with original
`pylint pylint.cfg all_modules` in all modules.
plus run with new conventions in just modified modules
`pylint pylint_pr.cfg modified_modules`
This is just to pylint cases.

moylop260 commented:
FYI 
![travis](https://cloud.githubusercontent.com/assets/6644187/9338365/fe06b3f8-45a9-11e5-8ce3-57b468d73813.png)

guewen commented:
Posted on the wrong thread before; so I post again here:

I missed #229 and I am not sure I would have agreed with in this form. I think these PRs are introducing a lot of complexity and I am not sure to get all the benefits they bring. I'm for keeping things simple and that seems to be too far. For instance, we decided previously not to support the differential diff for flake8 and I think we shouldn't support it for pylint neither.

pedrobaeza commented:
I think #229 change is a needed one to allow to do more things via python scripting, so I wouldn't discuss anything about it. Now, with this PR, I also have my doubts. @guewen, can you please explain why with flake8 we desestimate diff check?

moylop260 commented:
@guewen 
We don't like have a unique pylint-test to `differential diff`
We added a pylint-test-1 over all modules to check old conventions. (Same current functionallity).
We added a new pylint-test-2 to check new conventions but just to `differential diff`.
The focus is a little different.

Currently, we have a problem that if we add a new conventions in pylint-test-1 we will have all repos red in travis builds for stables branch and PR ones.

Then we need a middle step to add new conventions without have all repos red.
We can to change the modules progressively of new conventions.

moylop260 commented:
...after a period time we can to move from new conventions to old conventions.

guewen commented:
So we introduce a lot of new code, a lot of additional complexity only to handle the transition? Why not enforce the new rules starting from v9?

moylop260 commented:
@guewen 
Thank you for you comment.

Sorry if I wasn't clear.

I'm talking abount transition of old conventions to new conventions.
I don't talking about of v8 to v9

Currently we have next problem:
 1. We have created a new pr to oca/PROJECT-X branch 8.0
 2. I received feedabck from reviewers:
   3. Please, rename the class to `CamelCase`
   4. Please, add OCA author.
   5. Please, change your comment `encondig` by `coding`
   6. Other ones.

If we can automate it, Why not automate it?
Save time of reviewer, save time of developers if just in new changes.
We are working currently equal, but manually.

Tell me if I was more clear.

moylop260 commented:
NOTE: The code of this part just is here:
![code](https://cloud.githubusercontent.com/assets/6644187/9392349/b9295ab2-4741-11e5-88bb-59b300486394.png)

FYI You are in a progressive PR.

nhomar commented:
@moylop260 This PR is not mergeable can you rebase or close to discuss in the right place.

@guewen Complexity is a subjective word.

I think testing here is using the less complex tools availables "pylint plugin adding configuration to travis."

In other hands, this is the result of a work done due to the excesive waste of time of reviewers checking "lints" instead the content of modules itself, rounding PR revisions around CamelCase vs snake_case instead making it around the conceptualization and functionality itself.

Yes it introduce a lot of code but necesary code in only 1 place "the mqt" saving a lot of code in hundreds of repositories.

I am almost sure you are agreed on that.

About the second point and the diff-checking.

The diff checking to be clear is the "Modified Module" not the "Modified diff" it means if We use to fix Module A only this module will be checked in order to avoid introduce fake red in modules, but all teh repository will be tested as usual.

I do not know if it helps me to clarify the point?

Regards.

PS: THis is the huge issue of ask for separeated PR to solve only 1 problem just because it is big. If it is a Big functionality you must expect a BIG pr but , another history issue...

moylop260 commented:
@nhomar 

This is a progressive PR that depend of:
https://github.com/OCA/maintainer-quality-tools/pull/230

To review #232 PR you will need review just last commit diff.
I will to rebase when #230 is merged but now is blocked by @guewen because he is commenting in other one over this one.

lepistone commented:
I see your points @moylop260 @nhomar, and I really appreciate your efforts on the MQT. However, I also feel a bit uncomfortable about MQT becoming full-fledged complex scripts, because they are still coupled to a specific use case, and they need big adjustments just to run in different scenarios (see https://github.com/OCA/maintainer-quality-tools/pull/236#issuecomment-133319604 for example, or the need of the travis2docker script to run those locally). I would be probably happy enough with just adding new rules on v9.

Thanks for your time!

nhomar commented:
>> I would be probably happy enough with just adding new rules on v9.

@lepistone 

I am not totally on disagreed with this, but honestly for us this job is only to be sure avoid (and sorry for the french) the "pain in the ass" of being stopped several PR because a CamelCase.

BUt what happen, we have an Egg/Chicken issue, if we because of fear do not merge and manage this today we are simply moving the problem for the future.

As I said before, IF it is a rule, put it on lint and that's all! no brain message to avoid brake the rules.

If it will be for v9.0 "fine" but then just STOP everybody to ask for such useless stuff on v8.0 then.

It can not be a process which really make us as reviewers and contributors waste a lot of time.

Just discuss again the rule and decide.

IMHO, (if it was in my power) I TOTALLy would merge this and use it (we already migrated using this more that 500 modules in our repositories and more than 5 repositories private in order to prepare the release.

But seriouslly my friend (and you know it is not personal just too tired) but continue stopping PR because LINT useless (or not so useless) rules checked manually is simply the MOST stupid thing I ever seen in opensource.

Opensource is a matter of efficiency, not make everybody waste wrongly the time (that's what salesman dows in privative world).

We are agreed on the rule? cool, let's pylint it... "simply!"

regards.

guewen commented:
Do you have examples of PR being blocked for a `CamelCase`?

nhomar commented:
Dude! is serious this question?

I supose you use google for your email:

Simply make this search in your inbox:

"camelcase Pedro M. Baeza"

I found 43 mentions of such topic in 30 differents PR.

Search for:

"camelcase OCA"

I found 83 on 70 different PR.

That's enought?

Then you need to add the mails I deleted and the cases where simply the reviewer ignore it (extend it the problem for future v9.0).

Then my point "What can we do to avoid at least 83 comments and more of this type?" we developed a test, script to convert and set of rules to ensure it will not expend 1 single second on that anybody instead the programmer itself.

I hope it clarify my point.

nhomar commented:
@guewen @lepistone 

What I really do not understand, is comments about "fears" without technical comments on them.

We developed everything trying to use the highest standars and teh most secure tools, unit tests, pylint and so on..... 

Then... just because somebody do not understand a tool it does not mean it is dangerous or wrong!

Download it, test it, give technical feedback, but it does not add any value discuss rules already voted and approved in the mean of a PR it is not helpfull IMHO.

If you give me the use case which fears you, we develop the unit test that's the way, let's think how improve it.

In the past we had really huge horrible things which passed unseen just because we "trust" I understand that, everybody made mistakes, but try it... say "I think if you add this case or use this tool or check this stuff, on repo X the result was wrong or good let's improve it" it help more than "I fear about apply this".

guewen commented on travis/test_pylint:
```diff
@@ -9,12 +9,54 @@ import os
+    "travis_run_pylint_pr.cfg")
+
+is_pull_request = os.environ.get(
+    'TRAVIS_PULL_REQUEST', 'false') != 'false'
```
Why is `TRAVIS_PULL_REQUEST` needed?

moylop260 replied:
TRAVIS_PULL_REQUEST is a [environment variable of travis](http://docs.travis-ci.com/user/environment-variables/#Convenience-Variables)
![environment variables - travis ci 2015-08-21 09-41-33](https://cloud.githubusercontent.com/assets/6644187/9410958/d3e63378-47e8-11e5-85ce-b64515c24c43.png)
This is needed here because we need to know if is a PR to verify new conventions just in PR cases to detect a diff with base branch `TRAVIS_BRANCH`.
![environment variables - travis ci 2015-08-21 09-43-57](https://cloud.githubusercontent.com/assets/6644187/9411021/29e8fea4-47e9-11e5-82e1-25269c279a18.png)

Now we can get git diff of `TRAVIS_BUILD_DIR`
![environment variables - travis ci 2015-08-21 09-45-24](https://cloud.githubusercontent.com/assets/6644187/9411044/5d114688-47e9-11e5-8036-a53b2d53e20d.png)

With next command if is PR:
```python
  git --git_dir=${TRAVIS_BUILD_DIR}/.git diff-index origin/${TRAVIS_BRANCH}
```

Now, we have all items changed in a PR we need verify if is a folder and a module to add to paths-to-verify in pylint with new conventions (pylint conf pr file).

dreispt replied:
Why not stick to the 0/1 values like for other directives? Even better, why not do the check after a string eval to accept both 0/1 and false/true?

moylop260 replied:
We are using a variable of travis app... We don't have control in this one.
But I copied the use of this variable of other stable code in MQT:
[is_pull_request](https://github.com/OCA/maintainer-quality-tools/blob/87f2b269667c02af210493e929eb9524eb455907/travis/travis_run_tests#L57-L58)

guewen commented:
> What I really do not understand, is comments about "fears" without technical comments on them.

I gave few precise points: differential lint, usage of `click`

Also, this is simply the grow of the code, each time we want to support a new use case, the code grows and we should really ask us if we really need to support this use case. IMO the tools used for the builds should be as simple as possible, using the more as possible the existing tools at our disposal (flake8, pylint, ...) and the less as possible be a custom application.

I'm not saying that this work doesn't respect the quality standard, that it is badly done or anything. I see that @moylop260 did a lot of work here and it just makes me uncomfortable to come with my complaints. I'm not talking against the @moylop260's quality of code here, I just want to question the requirement in itself and also the fact which does not behove to him that we started to grow the MQT as a big application, which is slowly getting coupled to many many things. You should understand that my fears as you say are not on a particular line of code, but at how quickly the lines of code and the overall complexity grow in that project (and it doesn't only concern @moylop260 PRs). 

> Then... just because somebody do not understand a tool it does not mean it is dangerous or wrong!

No, but it could indicate that it is too complicated or not documented enough. If I can't understand it, chances are that the other ones neither, which is a bad sign for the future maintainability.

> "camelcase Pedro M. Baeza"

Maybe we should ask @pedrobaeza to relax on this rule? ;-) Personally, I won't block a PR because a class miss `CamelCase`. 

If we delay the usage of the new constraints to v9, we can spare 50 lines of code. That's my point.

pedrobaeza commented:
Hey, hey, I'm also not blocking any PR by CamelCase, but making code review, I find it and the OCA guidelines we all agreed ask for this convention, so this what I do, only asking for.

But as @nhomar and @moylop260 said before: if we can automate this (the check or maybe the conversion), why not do it? We already have it with 80 cols length, a lot of pylint rules, and so on.

This progressive changes are done to provide a good infrastructure to detect or expand these checks easily and to help in the deployment of these rules across all the OCA repositories. You were one of the first to complain when a new rule makes red a lot of repos. With this feature, we can make via .travis.yml in which repos it should be applied or not.

What is the problem then? Too much code? 50 lines more of code is too much code? We all deal everyday with thousands of code lines without problem, and having the code clearly commented and structured, this increase is not a problem. And I personally prefer 100 lines of Python code than 30 lines of ugly and weird bash shell script.

nhomar commented:
>> If we delay the usage of the new constraints to v9, we can spare 50 lines of code. That's my point.

I am not totally on disagreed on that as I said, but if the rule is that one, then let's automate it (even if it is not done today).

That's why @moylop260 tried to considere the case where the module which is being changed will be chacked "only" for this rule.

We lived already that on our own repositories with "eternal red's" becuase pylint stuff, then we changed the concept to test only the "proposed module" and it works like a charm, then moylop asked do such job for mqt and we think it will be usefull.

>> No, but it could indicate that it is too complicated or not documented enough. If I can't understand it, chances are that the other ones neither, which is a bad sign for the future maintainability.

Yes, this is a good technical reason, if you considere it has leak of documentation it is a good point, that's why I strongly asked some Issues Ago for the standarizations of mqt, trying to autobuild the documentation, use doctests and follow some standars on such topic....

it is not the same "explain" what a method does than explain with a working example.... 

There is a huge issue also (and I recognize the problem is the speed which we want to follow in some points) mixing several topics in the same PR, but sometimes duplicate the time expended in a job just because separate in one specific path a work is not good either,

Include click was a good Idea, and click is an excellent tool, then why not introduce such concept to save complexity of the code?.

Sometimes it is better if we pre-merge and then pre-test, "maybe creating a separated branch for interim time?"

but even me that knows what it does makes me really difficult review separatally this several PR related to the same topic I understand that.

moylop260 commented:
I had pr blocked in 8.0 because:
 - I dont used OCA author
 - I dont used coding comment
 - I dont used REAMDE.md
 And so on.
I dont talking about @pedrobaeza 
Im talking about of current reviewers.
I like this rules I like this block.
But if linter told me this fails is easielest for all.

moylop260 commented:
A note important, the branch v8 where I had pr blocked there are modules with this same fails.
Then this new conventions are currently progressive changed. Thats is coll to me too.
This PR allow add new conventions to a progressve work too

moylop260 commented on travis/cfg/travis_run_pylint_pr.cfg:
```diff
@@ -0,0 +1,33 @@
+[MASTER]
```
FYI this is a copy and paste of curreny cfg file but removing all enable checks.

moylop260 commented:
@guewen 

> Do you have examples of PR being blocked for a CamelCase? 

Example of cases:
 - Missing license key examples:
   - https://github.com/OCA/runbot-addons/pull/11/files#r34793801
 - Change `encoding` by `coding`
   - https://github.com/OCA/server-tools/pull/193#discussion_r35187469
 - Use Readme.rst file:
   - https://github.com/OCA/stock-logistics-workflow/pull/64/files#r24239000
 - CamelCase:
  - https://github.com/OCA/rma/pull/36#discussion_r36347320

moylop260 commented:
PR rebased.
Now we don't have many changes of progressive PR.
Can you review the diff please?

I will be happy to answer

moylop260 commented:
@guewen 
Do you have another comment, here?

pedrobaeza commented:
As this is only for pylint for style-questions, I see no harm and do benefits for allowing (via travis config, not by default) to only check modules that are affected by the PRs, instead of all. This way, we can adapt a new style rule, but not need to greenify all branches because they don't respect this rule, and progressive PRs will fix conventions for the touched modules.

So :+1:

nhomar commented:
:+1:

lmignon commented:
Self questioning...
 maintainer-quality-tools is one of the most sensitive repository in OCA. Changes here impact all the others repositories. Is the merging rules defined for others repositories well suited for this repository?

pedrobaeza commented:
Not really, @lmignon. Rules for now are the same. Anyways, this PR can help to make changes in this repo to be gradually propagated across repositories.

moylop260 commented:
I sent a email in community to told that we will enable CamelCase in MQT, 3
weeks ago.
This PR is part of this progressive change.
El ago 26, 2015 3:32 AM, "Pedro M. Baeza" <notifications@github.com>
escribió:

> Not really, @lmignon <https://github.com/lmignon>. Rules for now are the
> same. Anyways, this PR can help to make changes in this repo to be
> gradually propagated across repositories.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCA/maintainer-quality-tools/pull/232#issuecomment-134897541>
> .
>

moylop260 commented:
@gurneyalex @bwrsandman @dreispt Can you help me with your feedback here?

dreispt commented:
Question: could [running pylint on a diff](http://stackoverflow.com/questions/27486658/get-static-code-analyzer-messages-only-from-git-diff) be an alternate implementation?

dreispt commented on .travis.yml:
```diff
@@ -31,7 +31,8 @@ env:
   - VERSION="6.1" INCLUDE="test_module,second_module"
-  - LINT_CHECK="1" TESTS="0" PYLINT_EXPECTED_ERRORS="17"
+  - LINT_CHECK="1" TESTS="0" PYLINT_EXPECTED_ERRORS="17" TRAVIS_PULL_REQUEST="false" # Use main pylint config file
+  - LINT_CHECK="1" TESTS="0" PYLINT_EXPECTED_ERRORS="18" TRAVIS_PULL_REQUEST="true"  # Use PR pylint config file
```
IMO a better name would be `LINT_PULL_REQUEST`, and we usually use `0` or `1` shorter values (although there's nothing wrong with `true`/`false`).

moylop260 replied:
You are right!
However, We don't have control in env var from travis builds..
We are detecting if is a pull request with next variable:
http://docs.travis-ci.com/user/environment-variables/#Convenience-Variables

Similar to [transifex-script](https://github.com/OCA/maintainer-quality-tools/blob/87f2b269667c02af210493e929eb9524eb455907/travis/travis_run_tests#L57-L58)

dreispt replied:
This is not a TravisCI variable, it is added by us, so we can use whatever convention we want.

moylop260 replied:
Sorry, I'm talking about of TravisCI variable: `TRAVIS_PULL_REQUEST`
![travis_pr](https://cloud.githubusercontent.com/assets/6644187/9591467/21aafa74-5001-11e5-9959-42794e05375d.png)

I'm confunsing, can you help me to identify:
What `not-TravisCi` variable are you talking?

moylop260 replied:
@dreispt 
Do you another comment?
Can you help me with your feedback or plus one?

dreispt replied:
I think I got it now: you are overriding Travis variable `TRAVIS_PULL_REQUEST` to perform self tests...

dreispt commented on travis/run_pylint.py:
```diff
@@ -92,4 +92,8 @@ def main(paths, config_file, sys_paths=None, extra_params=None):
+        exit(main(standalone_mode=False))
+    except click.ClickException as e:
+        e.show()
+        exit(e.exit_code)
```
This usage does not justify adding the `click` dependency.

dreispt commented on travis/self_tests:
```diff
@@ -45,3 +45,6 @@ if os.environ.get('LINT_CHECK', 0) == 1:
     assert 1 == count_errors
+
+# Testing git run from getaddons
+getaddons.get_modules_changed(repo_dir)
```
Confusing. From a code reader POV, I expected the `get_modules_changed` result to be assigned to a list, that then would be used for a partial check...

moylop260 replied:
Ok, I got it

dreispt commented on travis/git_run.py:
```diff
@@ -0,0 +1,36 @@
+                   '--cached', base_ref]
+        res = self.run(command)
+        items = res.split('\n') if res else []
+        return items
```
Do we really need an Object here? I feel that in this case the code would be clearer if these methods were simple functions.

moylop260 replied:
This is for use:
```python
git_obj = GitRun(REPO_PATH)
git_obj.run(['fetch'])
git_obj.run(['checkout'])
git_obj.get_items_changed('8.0')
```
(This code is similar to runbot.)

Other case, we will need to add the global values in each method call, example:
```python
git_run.run(['git', '--repo-path=' + REPO_PATH, 'fetch'])
git_run.run(['git', '--repo-path=' + REPO_PATH, 'checkout'])
git_run.get_items_changed(REPO_PATH, '8.0')
```

IMHO The first one is clearer.
What do you think?

dreispt replied:
The second looks like you could just use `popen`directly.
What do you think of this:
```
git_run('fetch', path=REPO_PATH])
git_run('checkout', path=REPO_PATH])
git_get_items_changed(repo=REPO_PATH, branch='8.0')
```

moylop260 replied:
My vote to use:
```python
git_obj = git_run.GitRun(REPO_PATH)
git_obj.run(['fetch'])
git_obj.run(['checkout'])
git_obj.get_items_changed('8.0')
```

@pedrobaeza 
What do you think about?

moylop260 replied:
@dreispt 

I searched opinions on the Internet, and they all agree with OOP.
Object Oriented Programming supports:
 - inheritance
 - encapsulation
 - abstraction
 - polymorphism
...
Structured don't support.

[Referencia](http://freefeast.info/difference-between/difference-between-structured-programming-and-object-oriented-programming-structured-programming-vs-object-oriented-programming/)

Other spanish forums say: OOP is clearer than structured.

What do you think?

dreispt replied:
@moylop260 It's not. OOP has some good use cases, but in most cases is bloated. Plain function are simpler and easier to read when a stateful object is not really needed.

But that's a long discussion. There is nothing "wrong" with your OOP implementation and if boils down to programming style preference. So go ahead.

dreispt commented:
OK, I took the time to do a code review, have a couple of comments, but overall it looks like a good feature to me. It surely will make adding new lint checks less painful (for pylint only, for now...).

moylop260 commented:
@dreispt 
Thank for you comment and time man.
I will give reply soon

moylop260 commented:
@dreispt 
> Question: could [running pylint on a diff](http://stackoverflow.com/questions/27486658/get-static-code-analyzer-messages-only-from-git-diff) be an alternate implementation?

Yes, we are using a similar case.

We get diff with:
`git diff-index --name-only`
Later, we have a filter to get all odoo modules paths that changed and run pylint in modules_changed.

Just files changed is a good approach but the focus of this PR is that if you change a module you will need to fix all conventions in all module files.

oscarolar commented:
:+1:

moylop260 commented:
PR of 15 days ago with 3 approvals...

But we have a discuss opened (and abandoned) by @guewen and @lepistone (without -1)
We will wait this weekend before of merge.

guewen commented:
@moylop260 I won't block this PR.

lepistone commented:
I feel I don't have anything new to say. 

Please don't take my opinion badly: I just don't see a big enough win to compensate the added complexity and coupling, that's all. This is not because of specific technical things, not because of usage of a library or another, or specific mistakes.

After all the work and discussion that has been made, I won't block this. Thanks for your efforts!

I still believe that adding complexity to MQT has a price, and we'd need at least some documentation, changelogs and releases.

Specific example: someone has a red build somewhere and they have to try to understand hundred of lines of code and discussions and git logs just to understand what is going on.

moylop260 commented:
@lepistone and @guewen
Thank for comment

> someone has a red build somewhere and they have to try to understand
hundred of lines of code and discussions and git logs just to understand
what is going on.

You are right!
I will consider add a message very very specific in new checks (other PRs)
to know how fix it directly from message without need to read full code of
MQT.
El sep 3, 2015 6:47 AM, "Leonardo Pistone" <notifications@github.com>
escribió:

> I feel I don't have anything new to say.
>
> Please don't take my opinion badly: I just don't see a big enough win to
> compensate the added complexity and coupling, that's all. This is not
> because of specific technical things, not because of usage of a library or
> another, or specific mistakes.
>
> After all the work and discussion that has been made, I won't block this.
> Thanks for your efforts!
>
> I still believe that adding complexity to MQT has a price, and we'd need
> at least some documentation, changelogs and releases.
>
> Specific example: someone has a red build somewhere and they have to try
> to understand hundred of lines of code and discussions and git logs just to
> understand what is going on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCA/maintainer-quality-tools/pull/232#issuecomment-137414941>
> .
>

dreispt commented:
@moylop It would be nicer to ask for someone else to merge your own PR.

yvaucher commented:
+1 with @dreispt

moylop260 commented:
@dreispt
Thank for request.
I will take caution with merge my PRs
Can you send me the link with result of this discuss please?

dreispt commented:
Even if it's not a rule, my comment was out of common sense - MQT is a particularly sensitive repo and we extra care is needed.

But here is a relevant [link](https://odoo-community.org/groups/oca-contributors-15/oca-contributors-24573?mode=thread&date_begin=&date_end=).

moylop260 commented:
Ok I got it
Thank you.
Maybe, should be a good idea added in
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
El sep 4, 2015 8:44 AM, "Daniel Reis" <notifications@github.com> escribió:

> Even if it's not a rule, my comment was out of common sense - MQT is a
> particularly sensitive repo and we extra care is needed.
>
> But here is a relevant link
> <https://odoo-community.org/groups/oca-contributors-15/oca-contributors-24573?mode=thread&date_begin=&date_end=>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCA/maintainer-quality-tools/pull/232#issuecomment-137741649>
> .
>
